src/ucs/config/parser.c

```diff
@@ -855,6 +858,84 @@ void ucs_config_help_array(char *buf, size_t max, const void *arg)
     array->parser.help(buf + strlen(buf), max - strlen(buf), array->parser.arg);
 }
 
+int ucs_config_sscanf_allow_list(const char *buf, void *dest, const void *arg)
+{
+    ucs_config_allow_list_t *field  = dest;
+    unsigned offset                 = 0;
+
+    if (buf != NULL && buf[0] == ucs_config_parser_negate) {
+        field->mode = UCS_CONFIG_ALLOW_LIST_NEGATE;
+        offset++;
+    } else {
+        field->mode = UCS_CONFIG_ALLOW_LIST_ALLOW;
+    }
+
+    if (buf == NULL || !ucs_config_sscanf_array(&buf[offset], dest, arg)) {
+        return 0;
+    }
+
+    if (field->array.count >= 1 && !strcmp(field->array.names[0], UCS_CONFIG_PARSER_ALL)) {
+        field->mode = UCS_CONFIG_ALLOW_LIST_ALLOW_ALL;
+        ucs_assert(field->array.count == 1);
```

Comment:
maybe remove this assert?

Reply:
Why? Isn't `all,some1,some2` an invalid input?

Reply:
if it's invalid input we should return error rather than assert + gtest to check it returns err